pkg/pillar/cmd/zedrouter/pubsubhandlers.go

```diff
@@ -629,3 +631,159 @@ func (z *zedrouter) handleAppInstDelete(ctxArg interface{}, key string,
 	z.unpublishAppInstMetadata(appInstMetadata)
 	z.log.Functionf("handleAppInstDelete(%s) done", key)
 }
+
+func (z *zedrouter) handlePatchEnvelopeImpl(peInfo types.PatchEnvelopeInfoList) {
+	before := z.patchEnvelopes.GetUUIDList()
+	envelopes := MapFromPatchEnvelopeInfo(peInfo.Envelopes)
+	after := make([]uuid.UUID, 0, len(envelopes))
+	for k := range envelopes {
+		after = append(after, k)
+	}
+	z.patchEnvelopes.UpdateEnvelopes(envelopes)
+
+	toDelete, _ := generics.DiffSets(before, after)
+
+	if len(toDelete) > 0 {
+		go func() {
+			z.patchEnvelopes.DeleteStateNotificationCh() <- toDelete
+		}()
+	}
+	if len(after) > 0 {
+		go func() {
+			z.patchEnvelopes.UpdateStateNotificationCh() <- after
+		}()
```

Comment:
You are stlll spawning a potentially unbounded set of go routines. 

Can't you record the fact that a PE should be deleted in the PE itself and only send a empty notification (struct{}) to the channel that work is needed, and then the goroutine reading the channel will look at all of the PEs for the ones marked "to delete" and "to update"?